### ISSUE TITLE
[Charts Tooltip] Number of data series displayed should be defined at chart level, not at time bucket level

### ISSUE THREAD

**Issue by gvnmagni** [elastic/elastic-charts#2297]
With the introduction of the new tooltip, the number of data series displayed within that has been limited to a max of 10 elements, in case of higher numbers only the hovered data series is displayed.
This behavior is established at each time bucket but unfortunately this creates a slightly uncomfortable experience when moving from bucket to the other, as we can see in the following gif.

![Jan-12-2024 12-46-27](https://github.com/elastic/elastic-charts/assets/18484576/cfbc0b9a-c3f9-4777-b15d-50f674f68691)

I would suggest to determine the number of data series displayed at chart level so that all the buckets behave the same way.

**Comment by nickofthyme** [comment]
Yeah I agree, though this was deliberately done this way so I marked it as an enhancement rather than a bug.

**Comment by gvnmagni** [comment]
> Yeah I agree, though this was deliberately done this way so I marked it as an `enhancement` rather than a `bug`.

thank you Nick! 👍
